File: C#/BinarySearch.cs
// Author: Luis A. Tavarez - Country (DR - Dominican Republic) - 2024  
// Hacktoberfest 2024
// C#

using System;

class Program
{
    // Binary Search Algorithm
    static int BinarySearch(int[] arr, int target)
    {
        int start = 0;
        int end = arr.Length - 1;

        while (start <= end)
        {
            int mid = start + (end - start) / 2;

            // If the element is present at the middle
            if (arr[mid] == target)
            {
                return mid;
            }

            // If target is greater, ignore the left half
            if (arr[mid] < target)
            {
                start = mid + 1;
            }
            // If target is smaller, ignore the right half
            else
            {
                end = mid - 1;
            }
        }

        // Return -1 if the element is not present
        return -1;
    }

    static void Main(string[] args)
    {
        int[] array = { 2, 3, 4, 10, 40 };
        int target = 10;
        int result = BinarySearch(array, target);

        if (result != -1)
        {
            Console.WriteLine("Element found at index: " + result);
        }
        else
        {
            Console.WriteLine("Element not found.");
        }
    }
}

File: C++/Binary_search.cpp
#include <iostream>
using namespace std;

int binarySearch(int arr[], int n, int target) {
    int left = 0;
    int right = n - 1;

    while (left <= right) {
        int mid = left + (right - left) / 2; // Prevents overflow

        if (arr[mid] == target) {
            return mid; // Target found
        }
        if (arr[mid] < target) {
            left = mid + 1; // Search in the right half
        } else {
            right = mid - 1; // Search in the left half
        }
    }
    return -1; // Target not found
}

int main() {
    int arr[] = {1, 2, 3, 4, 5, 6, 7};
    int n = sizeof(arr) / sizeof(arr[0]);
    int target = 5;

    int result = binarySearch(arr, n, target);
    
    if (result != -1) {
        cout << "Element found at index: " << result << endl;
    } else {
        cout << "Element not found." << endl;
    }

    return 0;
}
